lighthouse-core/gather/gatherers/seo/font-size.js

```diff
@@ -144,107 +233,121 @@ function getFontSizeInformation(driver, node) {
-  return node.nodeType === global.Node.TEXT_NODE &&
+  return (
+    node.nodeType === TEXT_NODE_TYPE &&
     !TEXT_NODE_BLOCK_LIST.has(node.parentNode.nodeName) &&
-    getNodeTextLength(node) > 0;
+    getNodeTextLength(node) > 0
+  );
 }
 
 class FontSize extends Gatherer {
+  /**
+   * @param {LH.Gatherer.PassContext['driver']} driver
+   */
+  static async fetchNodesToAnalyze(driver) {
+    let failingTextLength = 0;
+    let visitedTextLength = 0;
+    let totalTextLength = 0;
+
+    const nodes = await getAllNodesFromBody(driver);
+
+    const textNodes = nodes.filter(isNonEmptyTextNode);
+    totalTextLength = textNodes.reduce((sum, node) => (sum += getNodeTextLength(node)), 0);
+
+    const nodesToVisit = textNodes
+      .sort((a, b) => getNodeTextLength(b) - getNodeTextLength(a))
+      .slice(0, MAX_NODES_VISITED);
+
+    const nodePromises = nodesToVisit.map(node => fetchComputedFontSize(driver, node));
+    const visitedNodes = await Promise.all(nodePromises);
+
+    /** @type {Array<FailingNodeData>} */
+    const failingNodes = [];
+    for (const visitedNode of visitedNodes) {
+      if (!visitedNode) continue;
+      visitedTextLength += visitedNode.textLength;
+
+      if (visitedNode.fontSize < MINIMAL_LEGIBLE_FONT_SIZE_PX) {
+        failingNodes.push(visitedNode);
+        failingTextLength += visitedNode.textLength;
+      }
+    }
+
+    const nodesToAnalyze = failingNodes
+      .sort((a, b) => b.textLength - a.textLength)
+      .slice(0, MAX_NODES_ANALYZED);
+
+    return {totalTextLength, visitedTextLength, failingTextLength, nodesToAnalyze};
+  }
+
+  /**
+   *
+   * @param {LH.Gatherer.PassContext['driver']} driver
+   * @param {Array<FailingNodeData>} nodesToAnalyze
+   */
+  static async analyzeFailingNodes(driver, nodesToAnalyze) {
+    const analysisPromises = nodesToAnalyze.map(async failingNode => {
+      failingNode.cssRule = await fetchSourceRule(driver, failingNode.node);
+      return failingNode;
+    });
+
+    const analyzedFailingNodesData = await Promise.all(analysisPromises);
+
+    const analyzedFailingTextLength = analyzedFailingNodesData.reduce(
+      (sum, {textLength}) => (sum += textLength),
+      0
+    );
+
+    return {analyzedFailingNodesData, analyzedFailingTextLength};
+  }
+
   /**
    * @param {LH.Gatherer.PassContext} passContext
    * @return {Promise<LH.Artifacts.FontSize>} font-size analysis
    */
-  afterPass(passContext) {
+  async afterPass(passContext) {
     /** @type {Map<string, LH.Crdp.CSS.CSSStyleSheetHeader>} */
     const stylesheets = new Map();
     /** @param {LH.Crdp.CSS.StyleSheetAddedEvent} sheet */
     const onStylesheetAdd = sheet => stylesheets.set(sheet.header.styleSheetId, sheet.header);
     passContext.driver.on('CSS.styleSheetAdded', onStylesheetAdd);
 
-    const enableDOM = passContext.driver.sendCommand('DOM.enable');
-    const enableCSS = passContext.driver.sendCommand('CSS.enable');
+    await passContext.driver.sendCommand('DOM.enable');
```

Comment:
not that it's a big deal but we had a Promise.all in the previous version. Probably not worth the hassle
`await Promise.all([passContext.driver.sendCommand('DOM.enable'), passContext.driver.sendCommand('CSS.enable')]);`

Reply:
sure might as well